src/Microsoft.AspNetCore.Razor/TagHelpers/DefaultTagHelperContent.cs

```diff
@@ -298,10 +292,8 @@ private bool IsEmptyOrWhiteSpaceCore(object entry, EmptyOrWhiteSpaceWriter write
             {
                 // Use NullHtmlEncoder to avoid treating encoded whitespace as non-whitespace e.g. "\t" as "&#x9;".
                 ((IHtmlContent)entry).WriteTo(writer, NullHtmlEncoder.Default);
-                if (!writer.IsEmptyOrWhiteSpace)
-                {
-                    return false;
-                }
+
+                return writer.IsEmptyOrWhiteSpace;
```

Comment:
Hmm, for consistency, I should either revert changes to this block or change the `if` side to `return string.IsNullOrWhiteSpace(stringValue);` and remove the final `return true;` Any preferences one way or the other @NTaylorMullen? (I lean toward making further changes but only because that allows me to reduce nesting -- no need for the `else` wrapper.)